gateway-server/src/main/java/org/apache/knox/gateway/GatewayServer.java

```diff
@@ -269,10 +280,28 @@ private static void validateKerberosConfig(GatewayConfig config) throws GatewayC
   }
 
   private static void checkIfFileExistsAndCanBeRead(File fileToBeChecked, String propertyName, Set<String> errors) {
+    checkIfFileExistsAndCanBeRead(fileToBeChecked, propertyName, errors, false);
+  }
+
+  private static void checkIfDirectoryExistsAndCanBeRead(File fileToBeChecked, String propertyName, Set<String> errors) {
```

Comment:
Since these are new methods can we pass in a `Path` instead of a `File`?

Reply:
Fixed